test/warp.jl

```diff
@@ -387,4 +387,40 @@ ref_img_pyramid_grid = Float64[
             @test nearlysame(round.(Float64.(imgrq_cntr[axes(imgrq_cntr)...]), digits=3), round.(ref_img_pyramid_quad, digits=3))
         end
     end
+
+    @testset "imrotate" begin
+        # imrotate is a wrapper, hence only need to guarantee the interface works as expected
+
+        test_types = (Float32, Float64, N0f8, N0f16)
+        graybar = repeat(range(0,stop=1,length=100),1,100)
+        @testset "interface" begin
+            for T in test_types
+                img = Gray{T}.(graybar)
+                @test_nowarn imrotate(img, 45)
+                @test_nowarn imrotate(img, 45, Constant())
+                @test_nowarn imrotate(img, 45, Linear())
+                @test_nowarn imrotate(img, 45, axes(img))
+                @test_nowarn imrotate(img, 45, axes(img), Constant())
```

Comment:
since you mention it in the signature of the docs, it would be useful to assert that

```julia
@test imrotate(img, pi) ≈ imrotate(img, pi, Linear(), NaN)
```

Reply:
By adding this test, I met one weird problem. Not even know how could this happen.

```julia
# For Float type only
img = Gray{Float32}.(ones(5,5))
a1 = imrotate(img, π/4)
a2 = imrotate(img, π/4, Linear())
isequal(a1, a2) # false
isequal(channelview(a1), channelview(a2)) # true
```

Hence if I want to let the test pass, I need to write
```julia
@test isequal(channelview(imrotate(img, pi)), channelview(imrotate(img, pi, Linear(), NaN)))
```

Reply:
that is strange. maybe leave a todo comment explaining this for some future investigation